### ISSUE TITLE
when kicked out to the usage message, tell me why

### ISSUE THREAD

**Issue by grimm26** [turnerlabs/fargate#56]
I was trying to use fargate service deploy with both the image and revision flags and it kept just erroring out to the usage message.  Nowhere did it say that I can't use those two options together.  Since the command was buried in a CodeBuild yaml it took me a while to figure out why it was failing.

**Comment by jritsema** [comment]
Hi @grimm26.  I'm sorry that you had trouble.  So if I'm understanding your scenario correctly, it sounds like you're trying to run something like:

```
fargate service deploy -i <image> -r <revision>
```

Is that right?

That combination of flags wouldn't seem to make sense as the `-i` flag registers a new task definition based on the image and deploys it to the service, while the `-r` flag says to deploy a specific existing task definition revision.  Note that if you want to use the `r` flag and deploy a specific task definition revision, you can also use the `fargate task register -i <image>` command to first register before deploying.

Regardless, I agree that the output messaging should be more clear.

**Comment by grimm26** [comment]
Right.  What I was trying to overcome was that I had a new task definition that existed that the current running task was not yet using.  I assumed that fargatecli would just use the latest task definition.  Since I found that it instead built a new task definition out of the one currently being used by the running task, I found he `--revision` flag and tried to use that to hint at which revision of the task definition to use.  Apparently, that option is not what I thought it was.  So instead, my codebuild job does a task definition deploy and then an image deploy.  Good times.
